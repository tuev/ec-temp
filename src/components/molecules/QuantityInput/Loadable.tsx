/**
 *
 * Asynchronously loads the component for QuantityInput
 *
 */

import loadable from 'utils/loadable'

export default loadable(() => import('./index'))
