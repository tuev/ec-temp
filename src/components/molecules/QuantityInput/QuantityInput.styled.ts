/**
 *
 * Styled for QuantityInput
 *
 */

import { createMuiTheme } from '@material-ui/core/styles'

export const theme = createMuiTheme({
  overrides: {
    MuiFab: {
      sizeSmall: {
        width: '36px',
        height: '36px',
      },
    },
    MuiOutlinedInput: {
      root: {
        borderRadius: '24px',
        margin: '0 8px',
        width: '72px',
      },
      input: {
        padding: '10px',
      },
    },
  },
})
