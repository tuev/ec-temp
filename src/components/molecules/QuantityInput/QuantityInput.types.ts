export type HandleQuantityChange = (value: number) => unknown

export interface IQuantityInput {
  value?: number
  onChange?: HandleQuantityChange
}

export type OnQuantityChange = (e: React.ChangeEvent<{}>) => unknown

export interface IQuantityHookHandle {
  increase: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
  decrease: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
}

export type QuantityHook = (
  value?: number,
  onChange?: HandleQuantityChange
) => [number, OnQuantityChange, IQuantityHookHandle]

export type QuantityProps = IQuantityInput
