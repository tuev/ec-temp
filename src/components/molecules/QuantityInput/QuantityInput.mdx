# Stories mdx for QuantityInput

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './QuantityInput.stories.tsx'
import QuantityInput from './QuantityInput.tsx'

<Story name="basic" />