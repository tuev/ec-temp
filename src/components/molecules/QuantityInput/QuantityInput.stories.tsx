/**
 *
 *  Stories for QuantityInput
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import QuantityInput from './QuantityInput'

export default {
  component: QuantityInput,
  includeStories: ['basic'],
  title: 'QuantityInput',
}

export const basic = () => (
  <div>
    <QuantityInput />
  </div>
)
