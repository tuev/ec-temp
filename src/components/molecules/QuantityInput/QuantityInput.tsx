/**
 *
 * QuantityInput
 *
 */

import React, { FC } from 'react'
import Wrapper from 'components/atoms/Wrapper'
import { Fab, TextField } from '@material-ui/core'
import { Add as AddIcon, Remove as MinusIcon } from '@material-ui/icons'
import { ThemeProvider } from '@material-ui/core/styles'
import { theme } from './QuantityInput.styled'
import useQuantityHook from './useQuantity'
import { QuantityProps } from './QuantityInput.types'

const QuantityInput: FC<QuantityProps> = props => {
  const { value, onChange }: QuantityProps = props
  const [
    quantity,
    handleQuantityChange,
    { increase, decrease },
  ] = useQuantityHook(value, onChange)
  return (
    <ThemeProvider theme={theme}>
      <Wrapper display="flex" alignItems="center">
        <Fab
          color="secondary"
          size="small"
          onClick={decrease}
          data-testid="decrease-quantity-btn"
        >
          <MinusIcon />
        </Fab>
        <TextField
          variant="outlined"
          type="number"
          inputProps={{ min: 0 }}
          value={quantity}
          onChange={handleQuantityChange}
        />
        <Fab
          color="secondary"
          size="small"
          onClick={increase}
          data-testid="increase-quantity-btn"
        >
          <AddIcon />
        </Fab>
      </Wrapper>
    </ThemeProvider>
  )
}

export default QuantityInput
