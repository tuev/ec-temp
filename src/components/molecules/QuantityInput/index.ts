/**
 *
 * Index for QuantityInput
 *
 */

import QuantityInput from './QuantityInput'

export default QuantityInput
