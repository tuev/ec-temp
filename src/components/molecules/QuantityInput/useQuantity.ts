import { useState, useCallback, useEffect } from 'react'
import { get } from 'lodash'
import { QuantityHook } from './QuantityInput.types'

const useQuantityHook: QuantityHook = (initValue, onChange) => {
  const [quantity, onQuantityChange] = useState<number>(initValue || 0)
  const handleQuantityChange = useCallback(e => {
    const valueOnchange = get(e, 'target.value', 0)
    onQuantityChange(Math.max(valueOnchange, 0))
  }, [])

  useEffect(() => {
    if (onChange) {
      onChange(quantity)
    }
  }, [onChange, quantity])

  const increase = useCallback(() => onQuantityChange(+quantity + 1), [
    quantity,
  ])
  const decrease = useCallback(
    () => onQuantityChange(Math.max(+quantity - 1, 0)),
    [quantity]
  )
  return [quantity, handleQuantityChange, { increase, decrease }]
}

export default useQuantityHook
