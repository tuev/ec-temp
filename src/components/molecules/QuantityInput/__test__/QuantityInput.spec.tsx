/**
 *
 * Tests for QuantityInput
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import QuantityInput from '..'

import { renderHook, act } from '@testing-library/react-hooks'
import useQuantity from '../useQuantity'

describe('App QuantityInput', () => {
  it('Test get default QuantityInput no params ', () => {
    // test snapshot
    const initValue = 1
    let quantityValue = initValue
    const changeValue = 4
    const handleQuantityChange = jest
      .fn()
      .mockImplementation(value => (quantityValue = value))

    const { result } = renderHook(() =>
      useQuantity(initValue, handleQuantityChange)
    )

    expect(result.current[0]).toEqual(quantityValue)
    act(() => result.current[1]({ target: { value: changeValue } }))
    expect(quantityValue).toEqual(changeValue)
    act(() => result.current[2].increase())
    expect(quantityValue).toEqual(changeValue + 1)

    act(() => result.current[2].decrease())
    expect(quantityValue).toEqual(changeValue)
  })

  it('Test get default QuantityInput no params ', () => {
    // test snapshot
    const wrapper = render(<QuantityInput />)
    expect(wrapper).toMatchSnapshot()
  })
})
