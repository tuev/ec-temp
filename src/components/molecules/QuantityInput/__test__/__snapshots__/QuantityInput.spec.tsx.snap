// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App QuantityInput Test get default QuantityInput no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<body>
    <div>
      <div
        class="c0"
        display="flex"
      >
        <button
          class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
          data-testid="decrease-quantity-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              min="0"
              type="number"
              value="0"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-61 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-62"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
          data-testid="increase-quantity-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm gbyRbq"
      display="flex"
    >
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
        data-testid="decrease-quantity-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            min="0"
            type="number"
            value="0"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-61 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-62"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
        data-testid="increase-quantity-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App QuantityInput Test get default QuantityInput with event handler  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
      <div
        class="c0"
        display="flex"
      >
        <button
          class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
          data-testid="decrease-quantity-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              min="0"
              type="number"
              value="5"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-135 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-136"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
          data-testid="increase-quantity-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm gbyRbq"
      display="flex"
    >
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
        data-testid="decrease-quantity-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13H5v-2h14v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            min="0"
            type="number"
            value="5"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-135 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-136"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
        data-testid="increase-quantity-btn"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
